File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.py
def solution(arr1, arr2):
    r=[]
    # arr1행의 길이만큼 돌면서
    for i in range(len(arr1)):
        # 2차원 배열 공간 추가
        r+=[[]]
        # arr2 열의 길이만큼 행렬 곱과 합을 해준 요소 넣어주기 위해 반복문 돌면서
        for k in range(len(arr2[0])):
            # 합을 넣을 임시 변수 초기화
            s=0
            '''
            arr1열의 길이와 arr2 행의 길이는 같으므로
            arr2행의 길이만큼 행렬 곱과 합 진행
            '''
            for j in range(len(arr1[i])):
                s+=(arr1[i][j]*arr2[j][k])
            # 합을 해당 배열 공간에 넣어줍니다.
            r[i]+=[s]
    return r

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    vector<vector<int>> r;
    vector<int> t;
    int s;
    for(int i=0;i<arr1.size();i++) {
        t={};
        for(int k=0;k<arr2[0].size();k++) {
            s=0;
            for(int j=0;j<arr1[i].size();j++) {
                s+=arr1[i][j]*arr2[j][k];
            }
            t.push_back(s);
        }
        r.push_back(t);
    }
    return r;
}